### PR TITLE
Enhanced RDoc for Enumerable#slice_before

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: enum.c [hunk 1]
```diff
@@ -3796,24 +3796,78 @@ slicebefore_i(RB_BLOCK_CALL_FUNC_ARGLIST(yielder, enumerator))
...
- *  If <code>_pattern_ === _elt_</code> returns <code>true</code> or the block
- *  returns <code>true</code> for the element, the element is beginning of a
- *  chunk.
-
- *  The <code>===</code> and _block_ is called from the first element to the last
- *  element of _enum_.  The result for the first element is ignored.
-
- *  The result enumerator yields the chunked elements as an array.
- *  So +each+ method can be called as follows:
+ *  Returns an Enumerator.
```
Comment (jeremyevans) [inline enum.c:3802]: I think we can remove this line.  It's obvious from the call-seq.

File: enum.c [hunk 2]
```diff
@@ -3796,24 +3796,78 @@ slicebefore_i(RB_BLOCK_CALL_FUNC_ARGLIST(yielder, enumerator))
...
+ *
+ *   e = (1..15).slice_before(6) # => #<Enumerator: ...>
+ *   e.each {|array| p array }
+ *
+ *  Output:
+ *
+ *    [1, 2, 3, 4, 5]
+ *    [6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+ *
+ *  Example with an array and string pattern:
```
Comment (jeremyevans) [inline enum.c:3819]: I don't think we need 4 examples of pattern argument usage, as they are more of an example of differences in `===` behavior than `slice_before` behavior.  A single example should suffice, pick your favorite.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for working on this.  Please see requested changes.

File: enum.c (hunk 2 above)
Comment (BurdetteLamar) [reply enum.c]: Done.

**Pull request merged**